### PR TITLE
[Test] Bump mono and use the new splited tests.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Bump mono to get the new splited test dlls and add them to be ran in
xharness. Special logic is used for mscorlib so we make sure that all
the 'parts' of the test dll do have the same configurations.

### CODE REVIEW

File: tests/xharness/BCLTestImporter/BCLTestImportTargetFactory.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ public List<MacTestProject> GetMacBclTargets (MacFlavors flavor)
 			var result = new List<MacTestProject> ();
 			foreach (var tp in projectGenerator.GenerateAllMacTestProjects (platform)) {
 				var prefix = tp.XUnit ? "xUnit" : "NUnit";
-				var finalName = (tp.Name == "mscorlib") ? tp.Name : $"[{prefix}] Mono {tp.Name}"; // mscorlib is our special test
+				var finalName = tp.Name.StartsWith("mscorlib", StringComparison.Ordinal) ? tp.Name : $"[{prefix}] Mono {tp.Name}"; // mscorlib is our special test
```
Comment (wachaudh) [inline tests/xharness/BCLTestImporter/BCLTestImportTargetFactory.cs]:
```suggestion
				var finalName = tp.Name.StartsWith ("mscorlib", StringComparison.Ordinal) ? tp.Name : $"[{prefix}] Mono {tp.Name}"; // mscorlib is our special test
```

File: tests/xharness/BCLTestInfo.cs [hunk 2]
```diff
@@ -122,7 +122,7 @@ protected void Process (string test_sources, IEnumerable<string> test_files, str
 
 		public virtual void Convert ()
 		{
-			var testName = TestName == "mscorlib" ? "corlib" : TestName;
+			var testName = TestName.StartsWith("mscorlib", StringComparison.Ordinal) ? "corlib" : TestName;
```
Comment (wachaudh) [inline tests/xharness/BCLTestInfo.cs]:
```suggestion
			var testName = TestName.StartsWith ("mscorlib", StringComparison.Ordinal) ? "corlib" : TestName;
```

File: tests/xharness/BCLTestInfo.cs [hunk 3]
```diff
@@ -202,7 +202,7 @@ public override void Convert ()
 
 		string GetFileList ()
 		{
-			var testName = TestName == "mscorlib" ? "corlib" : TestName;
+			var testName = TestName.StartsWith("mscorlib", StringComparison.Ordinal) ? "corlib" : TestName;
```
Comment (wachaudh) [inline tests/xharness/BCLTestInfo.cs]:
```suggestion
			var testName = TestName.StartsWith ("mscorlib", StringComparison.Ordinal) ? "corlib" : TestName;
```

File: tests/xharness/BCLTestImporter/BCLTestImportTargetFactory.cs [hunk 4]
```diff
@@ -37,12 +37,12 @@ public List<iOSTestProject> GetiOSBclTargets ()
...
 			foreach (var tp in projectGenerator.GenerateAlliOSTestProjects ()) {
 				var prefix = tp.XUnit ? "xUnit" : "NUnit";
-				var finalName = (tp.Name == "mscorlib") ? tp.Name : $"[{prefix}] Mono {tp.Name}"; // mscorlib is our special test
+				var finalName = tp.Name.StartsWith("mscorlib", StringComparison.Ordinal) ? tp.Name : $"[{prefix}] Mono {tp.Name}"; // mscorlib is our special test
 				result.Add (new iOSTestProject (tp.Path) {
 					Name = finalName,
 					SkipiOSVariation = !tp.Platforms.Contains (Platform.iOS),
 					SkiptvOSVariation = !tp.Platforms.Contains (Platform.TvOS),
-					SkipwatchOS32Variation = tp.Name == "mscorlib", // mscorlib is our special test
+					SkipwatchOS32Variation = tp.Name.StartsWith("mscorlib", StringComparison.Ordinal), // mscorlib is our special test
```
Comment (wachaudh) [inline tests/xharness/BCLTestImporter/BCLTestImportTargetFactory.cs]:
```suggestion
					SkipwatchOS32Variation = tp.Name.StartsWith ("mscorlib", StringComparison.Ordinal), // mscorlib is our special test
```

File: tests/xharness/BCLTestImporter/BCLTestImportTargetFactory.cs [hunk 5]
```diff
@@ -37,12 +37,12 @@ public List<iOSTestProject> GetiOSBclTargets ()
 			// generate all projects, then create a new iOSTarget per project
 			foreach (var tp in projectGenerator.GenerateAlliOSTestProjects ()) {
 				var prefix = tp.XUnit ? "xUnit" : "NUnit";
-				var finalName = (tp.Name == "mscorlib") ? tp.Name : $"[{prefix}] Mono {tp.Name}"; // mscorlib is our special test
+				var finalName = tp.Name.StartsWith("mscorlib", StringComparison.Ordinal) ? tp.Name : $"[{prefix}] Mono {tp.Name}"; // mscorlib is our special test
```
Comment (wachaudh) [inline tests/xharness/BCLTestImporter/BCLTestImportTargetFactory.cs]:
```suggestion
				var finalName = tp.Name.StartsWith ("mscorlib", StringComparison.Ordinal) ? tp.Name : $"[{prefix}] Mono {tp.Name}"; // mscorlib is our special test
```

File: tools/bcl-test-importer/BCLTestImporter/BCLTestProjectGenerator.cs [hunk 6]
```diff
@@ -135,10 +135,11 @@ public class BclTestProjectInfo {
 			new BclTestProjectInfo { Name = "SystemNetHttpFunctionalTestsXunit", assemblies = new [] { "monotouch_System.Net.Http.FunctionalTests_xunit-test.dll" }, Group = "BCL tests group 5" },
 
 			// Special assemblies that are in a single application due to their size being to large for the iOS 32b.
-			new BclTestProjectInfo { Name = "mscorlib", assemblies = new [] { "monotouch_corlib_xunit-test.dll" }, Group = "mscorlib" }, // special testcase for the corlib which is later used in xHarness for diff config options
-			new BclTestProjectInfo { Name = "SystemCoreXunit", assemblies = new [] { "monotouch_System.Core_xunit-test.dll" }, Group = "SystemCoreXunit" }, // special test case, this dll is to large for the iOS 32b
+			new BclTestProjectInfo { Name = "mscorlib Part 1", assemblies = new [] { "monotouch_corlib_xunit-test.part1.dll" }, Group = "mscorlib Part 1" }, // special testcase for the corlib which is later used in xHarness for diff config options
```
Comment (wachaudh) [inline tools/bcl-test-importer/BCLTestImporter/BCLTestProjectGenerator.cs]:
```suggestion
			new BclTestProjectInfo { Name = "mscorlib Part 1", assemblies = new [] { "monotouch_corlib_xunit-test.part1.dll" }, Group = "mscorlib Part 1" }, // special test case for the corlib which is later used in xHarness for diff config options
```

**Comment by mandel-macaque** [comment]
@wachaudh I'll apply the suggestions once we have results form the device tests, don't want to trigger a new build.

**Comment by mandel-macaque** [comment]
build

**Comment by mandel-macaque** [comment]
Bots had issues with the 32b dialog. Should have been fixed.

**Comment by mandel-macaque** [comment]
build

**Comment by mandel-macaque** [comment]
@monojenkins backport d16-5

**Pull request merged**